### PR TITLE
♻️🎨 `ooil config init` to create first`.osparc` config layout

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

`ooil` CLI can now create an `.osparc` config folder with `ooil config init` by running the [cookiecutter-osparc-service)](https://github.com/ITISFoundation/cookiecutter-osparc-service). In addition we do some refactoring of the oldest part of the tool before the coming upgrades

### Highlights

- 🎨  Extends run [cookiecutter-osparc-service)](https://github.com/ITISFoundation/cookiecutter-osparc-service) from CLI. To trigger
```
ooil config init
```
or
```
ooil config init --template=gh:pcrespov/cookiecutter-osparc-service --checkout=is1404/new-version_display
```
- ♻️  CLI logic 
    - Moves to `service_integration.cli`
    - Upgrades types Annotations
    - Improved doc

## Related issue/s

- Part of https://github.com/ITISFoundation/osparc-issues/issues/1404

## How to test

Manual testing using `ooil` image
```
cd packages/service-integration
make build
script/ooil.bash --help
```
image

## Dev-ops checklist
None

### CODE REVIEW

**Review by GitHK — APPROVED** [review]

File: packages/service-integration/src/service_integration/cli/_compose_spec.py [hunk 1]
```diff
@@ -109,9 +108,9 @@ def create_docker_compose_image_spec(
                 "No explicit config for OCI/label-schema found (optional), skipping OCI annotations."
             )
     # add required labels
-    extra_labels[f"{LS_LABEL_PREFIX}.build-date"] = datetime.utcnow().strftime(
-        "%Y-%m-%dT%H:%M:%SZ"
-    )
+    extra_labels[f"{LS_LABEL_PREFIX}.build-date"] = datetime.datetime.now(
+        datetime.timezone.utc
+    ).strftime("%Y-%m-%dT%H:%M:%SZ")
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/service-integration/src/service_integration/cli/_compose_spec.py:113]: Q: Why not use isoformat instead of ad-hoc formatting?

**Review by sanderegg — APPROVED** [review]

File: packages/service-integration/src/service_integration/cli/__init__.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    # save states
+    ctx.settings = AppSettings.parse_obj(overrides)
+
+
+#
+# REGISTER commands and/or sub-apps
+#
+
+app.command("compose")(_compose_spec.create_compose)
+app.add_typer(config_app, name="config", help="Manage osparc config files")
```
Comment (elisabettai) [inline packages/service-integration/src/service_integration/cli/__init__.py:67]: Minor: I don't see this helper in the screenshot of the command line interface. Shouldn't it be there?

Maybe the screenshot is not up-to-date or I am misunderstanding. 😊 

image

**Review by elisabettai — APPROVED** [review]

File: packages/service-integration/src/service_integration/cli/__init__.py (hunk 2 above)
Comment (pcrespov) [reply packages/service-integration/src/service_integration/cli/__init__.py:67]: old version 
image

File: packages/service-integration/src/service_integration/cli/_compose_spec.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcrespov) [reply packages/service-integration/src/service_integration/cli/_compose_spec.py:113]: Needs to follow RFC-3339

https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys

**Pull request merged**